File: appledoc/Extensions/NSFileManager+Appledoc.m
//
//  NSFileManager+Appledoc.m
//  appledoc
//
//  Created by Tomaz Kragelj on 1.11.12.
//  Copyright (c) 2012 Tomaz Kragelj. All rights reserved.
//

#import "NSFileManager+Appledoc.h"

@implementation NSFileManager (Appledoc)

- (BOOL)gb_fileExistsAndIsFileAtPath:(NSString *)path {
	BOOL isDirectory = NO;
	BOOL exists = [self fileExistsAtPath:path isDirectory:&isDirectory];
	return (exists && !isDirectory);
}

- (BOOL)gb_fileExistsAndIsDirectoryAtPath:(NSString *)path {
	BOOL isDirectory = NO;
	BOOL exists = [self fileExistsAtPath:path isDirectory:&isDirectory];
	return (exists && isDirectory);
}

@end

File: appledoc/Common/Extensions/NSFileManager+Appledoc.swift
//
//  Created by Tomaz Kragelj on 16.11.15.
//  Copyright © 2015 Gentle Bytes. All rights reserved.
//

import Foundation

extension FileManager {

	/** Determines whether the given path exists and represents a directory.
	
	@param path Path to examine.
	@return Returns true if path exists and is directory, false otherwise.
	*/
	func isPathDirectory(_ path: String) -> Bool {
		var isDirectory = ObjCBool(false)
		
		if !fileExists(atPath: path, isDirectory: &isDirectory) {
			return false
		}
		
		return isDirectory.boolValue
	}
	
}
